index.bs

```diff
@@ -606,6 +701,39 @@ SessionResult = (StatusResult)
 
 ```
 
+<div algorithm>
+
+To <dfn lt="updating the event map">update the event map</dfn>, given a |list of
+event names|, a |list of contexts|, and a |target status|:
+
+  1. Let |events| be an empty set.
+
+    1. For each entry |event name| in the |list of event names|,
+       let |events| be the union of |events| and the result of [=trying=] to
+       [=obtain a set of events from a name=] with |event name|.
+
+    1. If the |list of contexts| is null, let |targets| be a set containing the [=global
+       event set=]. Otherwise for each entry |context id| in the |list of
+       contexts|, run the following substeps:
+
+      1. If there is no browsing context with [=browsing context id=] |context id| return
+         [=error=] with [=error code=] [=invalid argument=]
+
+      1. Let |context| be the browsing context with id |context id|. If the
+         [=browsing context event map=] does not contain an entry for |context|,
+         set the value of the entry for |context| to a new empty map.
```

Comment:
I think the "create if not already there" style can be avoided here. Can't we just say that each target (or browsing context) has an event map which is initially an empty map, and never create it?

Reply:
Hanging custom state off objects we don't control in other specs is *probably*  OK, but I'm not sure it's worth the magic to save a few words (ideally the infra spec would give us better primitives here).

Reply:
Alright, I'm fine leaving this as is.